lalrpop/src/normalize/macro_expand/mod.rs

```diff
@@ -53,20 +53,41 @@ impl MacroExpander {
     }
 
     fn expand(&mut self, items: &mut Vec<GrammarItem>) -> NormResult<()> {
-        let mut counter = 0;
+        let mut counter = 0; // Number of items
+        let mut loop_counter = 0;
+
         loop {
             // Find any macro uses in items added since last round and
             // replace them in place with the expanded version:
             for item in &mut items[counter..] {
                 self.replace_item(item);
             }
             counter = items.len();
+            loop_counter += 1;
 
             // No more expansion to do.
             if self.expansion_stack.is_empty() {
                 return Ok(());
             }
 
+            if loop_counter > 200 {
```

Comment:
One way to somewhat alleviate the magic value here would for this to be the default, but have this be override-able via `Configuration`. What do you think?

Reply:
Yeah, this is a good idea.  I've added the config knob and pushed.